pydantic/functional_serializers.py

```diff
@@ -273,7 +270,7 @@ def serialize_courses_in_order(self, courses: Set[str]):
     """
 
     def dec(
-        f: Callable[..., Any] | staticmethod[Any, Any] | classmethod[Any, Any, Any],
```

Comment:
Are we losing specificity here?

Reply:
When type checking the body of `field_serializer`, type checkers will use the signature of that function (they won't use the overloaded signatures).

Previous to my changes, the return type was too broad (`Callable[[Any], Any]`). the argument type `Any` would match the type hint of `f` here, and the return type `Any` would also match `PydanticDescriptorProxy[Any]`.

Nothing we can really do here as we lie about the return type because we don't want to expose the `PydanticDescriptorProxy` class.